src/schema/v2/homeView/helpers/isSectionDisplayable.ts

```diff
@@ -27,5 +27,10 @@ export function isSectionDisplayable(
     })
   }
 
+  // section's display pre-check
+  if (typeof section.shouldBeDisplayed === "function") {
+    isDisplayable = isDisplayable && section?.shouldBeDisplayed(context)
+  }
```

Comment:
This comes after the other checks (auth, feature flags) because this one is potentially the most expensive, and we'd prefer to avoid computing it unless necessary